src/gpuarray/buffer_collectives.h

```diff
@@ -31,47 +31,43 @@ enum _gpucomm_reduce_ops {
   GA_SUM = 0,
   GA_PROD = 1,
   GA_MAX = 2,
-  GA_MIN = 3
+  GA_MIN = 3,
 };
 
 #define GA_COMM_ID_BYTES 128  // sizeof(gpucommCliqueId)
 /**
  * Dummy struct to define byte-array's length through a type
  */
 typedef struct _gpucommCliqueId {
-  char internal[GA_COMM_ID_BYTES];
+  char (*internal)[][GA_COMM_ID_BYTES];
```

Comment:
Why make the internal a pointer to a char table?  This also makes the struct 4 to 8 times bigger.